packages/block-library/src/navigation-menu-item/edit.js

```diff
@@ -1,93 +1,80 @@
-/**
- * External dependencies
- */
-import { invoke } from 'lodash';
-
 /**
  * WordPress dependencies
  */
 import { withSelect } from '@wordpress/data';
 import {
-	Dropdown,
 	ExternalLink,
-	IconButton,
 	PanelBody,
 	TextareaControl,
 	TextControl,
+	Toolbar,
 	ToggleControl,
+	ToolbarButton,
 } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import {
+	BlockControls,
 	InnerBlocks,
 	InspectorControls,
 	PlainText,
+	URLPopover,
 } from '@wordpress/block-editor';
 import {
 	Fragment,
-	useCallback,
 	useRef,
+	useState,
 } from '@wordpress/element';
 
-/**
- * Internal dependencies
- */
-import MenuItemActions from './menu-item-actions';
-const POPOVER_PROPS = { noArrow: true };
-
 function NavigationMenuItemEdit( {
 	attributes,
-	clientId,
 	isSelected,
 	isParentOfSelectedBlock,
 	setAttributes,
 } ) {
 	const plainTextRef = useRef( null );
-	const onEditLableClicked = useCallback(
-		( onClose ) => () => {
-			onClose();
-			invoke( plainTextRef, [ 'current', 'textarea', 'focus' ] );
-		},
-		[ plainTextRef ]
-	);
+	const [ isLinkOpen, setLinkOpen ] = useState( false );
+	const { label, url } = attributes;
 	let content;
 	if ( isSelected ) {
 		content = (
 			<div className="wp-block-navigation-menu-item__edit-container">
 				<PlainText
 					ref={ plainTextRef }
 					className="wp-block-navigation-menu-item__field"
-					value={ attributes.label }
-					onChange={ ( label ) => setAttributes( { label } ) }
+					value={ label }
+					onChange={ ( labelValue ) => setAttributes( { label: labelValue } ) }
 					aria-label={ __( 'Navigation Label' ) }
 					maxRows={ 1 }
 				/>
-				<Dropdown
-					contentClassName="wp-block-navigation-menu-item__dropdown-content"
-					position="bottom left"
-					popoverProps={ POPOVER_PROPS }
-					renderToggle={ ( { isOpen, onToggle } ) => (
-						<IconButton
-							icon={ isOpen ? 'arrow-up-alt2' : 'arrow-down-alt2' }
-							label={ __( 'More options' ) }
-							onClick={ onToggle }
-							aria-expanded={ isOpen }
-						/>
-					) }
-					renderContent={ ( { onClose } ) => (
-						<MenuItemActions
-							clientId={ clientId }
-							destination={ attributes.destination }
-							onEditLableClicked={ onEditLableClicked( onClose ) }
-						/>
-					) }
-				/>
 			</div>
 		);
 	} else {
-		content = attributes.label;
+		content = label;
 	}
 	return (
 		<Fragment>
+			<BlockControls>
+				<Toolbar>
+					<ToolbarButton
```

Comment:
I'm working on the new Link UI implementation. It's so common to want a button to trigger the link UI that I'm thinking we'll end up having a dedicated button component which we will expose for this use case.

Something like `<LinkControl.Toggle />`.

Reply:
I think it makes sense to have a toolbar link component that includes toggle and popover. As it is I'm repeating logic from the other instances of toolbar links, which is less than ideal.